src/confirm.rs

```diff
@@ -288,9 +288,17 @@ impl<T: Transport> Future for SendTransactionWithConfirmation<T> {
                     SendTransactionWithConfirmationState::GetTransactionReceipt(receipt_future)
                 }
                 SendTransactionWithConfirmationState::GetTransactionReceipt(ref mut future) => {
-                    let receipt = try_ready!(Future::poll(future))
-                        .expect("receipt can't be null after wait for confirmations; qed");
-                    return Ok(receipt.into());
+                    let err = "receipt can't be null after wait for confirmations; qed";
+                    if self.confirmations == 0 {
+                        match Future::poll(future)? {
+                            Async::Ready(opt_receipt) => return Ok(opt_receipt.expect(err).into()),
+                            Async::NotReady => SendTransactionWithConfirmationState::GetTransactionReceipt(*future),
```

Comment:
Using `ref mut future` is unavoidable in the arguments of the match arms, but dereferencing `*future` fails because there is no `Copy` for its type, and it is not feasible to do so.

Any suggestions appreciated.

Reply:
I don’t know how to satisfy the borrow checker here either. However, I doubt that even if this would compile this accomplishes what you want. With this implementation the code will effectively loop and block until a receipt is received. First this is counter to how futures work, and second this will not re-fetch a receipt and solve the issue.